File: api/src/main/java/fr/huiitre/tools/modules/riot/valorant/api/ValorantStoreHistoryController.java
package fr.huiitre.tools.modules.riot.valorant.api;

import fr.huiitre.tools.modules.riot.valorant.application.user.command.AddSkinToStoreHistoryCommand;
import fr.huiitre.tools.modules.riot.valorant.application.user.usecase.AddSkinToStoreHistoryUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.user.usecase.GetMyValorantStoreHistoryUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.user.view.ValorantStoreHistoryView;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/riot/valorant/store-history")
public class ValorantStoreHistoryController {

    private final GetMyValorantStoreHistoryUseCase getMyValorantStoreHistoryUseCase;
    private final AddSkinToStoreHistoryUseCase addSkinToStoreHistoryUseCase;

    public ValorantStoreHistoryController(
            GetMyValorantStoreHistoryUseCase getMyValorantStoreHistoryUseCase,
            AddSkinToStoreHistoryUseCase addSkinToStoreHistoryUseCase) {
        this.getMyValorantStoreHistoryUseCase = getMyValorantStoreHistoryUseCase;
        this.addSkinToStoreHistoryUseCase = addSkinToStoreHistoryUseCase;
    }

    @GetMapping
    public List<ValorantStoreHistoryView> getMyStoreHistory(@RequestParam Long accountId) {
        return getMyValorantStoreHistoryUseCase.execute(accountId);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public void addToStoreHistory(@RequestBody AddSkinToStoreHistoryCommand command) {
        addSkinToStoreHistoryUseCase.execute(command);
    }
}

File: api/Modules/Riot/Valorant/Api/ValorantStoreHistoryController.cs
using Microsoft.AspNetCore.Mvc;
using Tools.Api.Modules.Riot.Valorant.Application.User.Commands;
using Tools.Api.Modules.Riot.Valorant.Application.User.Usecases;
using Tools.Api.Modules.Riot.Valorant.Application.User.Views;

namespace Tools.Api.Modules.Riot.Valorant.Api;

// Historique des rotations de boutique, par compte Valorant lié.
[ApiController]
[Route("riot/valorant/store-history")]
public class ValorantStoreHistoryController : ControllerBase
{
    [HttpGet]
    public Task<List<ValorantStoreHistoryView>> GetMyStoreHistory(
        [FromServices] GetMyValorantStoreHistoryUseCase getMyValorantStoreHistoryUseCase,
        [FromQuery] long accountId
    )
    {
        return getMyValorantStoreHistoryUseCase.Execute(accountId);
    }

    [HttpPost]
    public async Task<IActionResult> AddToStoreHistory(
        [FromServices] AddSkinToStoreHistoryUseCase addSkinToStoreHistoryUseCase,
        [FromBody] AddSkinToStoreHistoryCommand command
    )
    {
        await addSkinToStoreHistoryUseCase.Execute(command);

        return StatusCode(StatusCodes.Status201Created);
    }
}
